From b47f26dd1f0993b589f3dd318d2a1878c62fd534 Mon Sep 17 00:00:00 2001
From: Shunqian Zheng <zhengsq@rock-chips.com>
Date: Fri, 29 Jun 2018 17:42:03 +0800
Subject: [PATCH] ARM: dts: rockchip: enable i2s/codec for px3se-evb

Px3se-evb board has two codecs es8396 and rk312x-codec.
ES8396 is enable by default. One can switch to rk312x-codec
via hardware rework: R7007, R7005, R7214, R7215.

This adds both es8396 and rk312x-codec but makes the latter one
disabled.

Change-Id: I51cad8bc50da8500c527d10ed52465c8f55d77c7
Signed-off-by: Shunqian Zheng <zhengsq@rock-chips.com>
---
 arch/arm/boot/dts/px3se-evb.dts | 76 +++++++++++++++++++++++++++++++++
 1 file changed, 76 insertions(+)

diff --git a/arch/arm/boot/dts/px3se-evb.dts b/arch/arm/boot/dts/px3se-evb.dts
index f9cd97224799..093251b3e30c 100644
--- a/arch/arm/boot/dts/px3se-evb.dts
+++ b/arch/arm/boot/dts/px3se-evb.dts
@@ -113,6 +113,45 @@
 		reset-gpios = <&gpio1 3 GPIO_ACTIVE_LOW>;
 	};
 
+	sound-rk312x {
+		compatible = "simple-audio-card";
+		status = "disabled";
+
+		simple-audio-card,format = "i2s";
+		simple-audio-card,mclk-fs = <256>;
+		simple-audio-card,name = "rockchip,rk312x-codec";
+		simple-audio-card,cpu {
+			sound-dai = <&i2s_8ch>;
+		};
+		simple-audio-card,codec {
+			sound-dai = <&codec>;
+		};
+	};
+
+	sound-es8396 {
+		compatible = "simple-audio-card";
+		status = "okay";
+
+		simple-audio-card,format = "i2s";
+		simple-audio-card,name = "rockchip,es8396-codec";
+		simple-audio-card,mclk-fs = <256>;
+		simple-audio-card,widgets =
+			"Microphone", "Microphone Jack",
+			"Line", "Microphone Headset",
+			"Headphone", "Headphone Jack";
+		simple-audio-card,routing =
+			"MIC", "Microphone Jack",
+			"DMIC", "Microphone Headset",
+			"Headphone Jack", "LOUTP",
+			"Headphone Jack", "ROUTN";
+		simple-audio-card,cpu {
+			sound-dai = <&i2s_2ch>;
+		};
+		simple-audio-card,codec {
+			sound-dai = <&es8396>;
+		};
+	};
+
 	usb_control {
 		compatible = "rockchip,rk3126-usb-control";
 
@@ -271,6 +310,20 @@
 	};
 };
 
+&codec {
+	#sound-dai-cells = <0>;
+	hp-ctl-gpios = <&gpio3 20 GPIO_ACTIVE_HIGH>;
+	spk-ctl-gpios = <&gpio0 30 GPIO_ACTIVE_HIGH>;
+	spk-mute-delay = <200>;
+	hp-mute-delay = <100>;
+	spk_volume = <25>;
+	hp_volume = <25>;
+	capture_volume = <26>;
+	gpio_debug = <1>;
+	codec_hp_det = <1>;
+	status = "disabled";
+};
+
 &cpu0 {
 	cpu-supply = <&vdd_arm>;
 };
@@ -516,6 +569,29 @@
 			};
 		};
 	};
+
+	es8396: es8396@11 {
+		status = "okay";
+		#sound-dai-cells = <0>;
+		compatible = "es8396";
+		reg = <0x11>;
+		clock-names = "mclk";
+		clocks = <&cru SCLK_I2S_OUT>;
+		spk-con-gpio = <&gpio0 30 GPIO_ACTIVE_HIGH>;
+		lineout-con-gpio = <&gpio3 20 GPIO_ACTIVE_HIGH>;
+	};
+};
+
+&i2s_2ch {
+	pinctrl-names = "default";
+	pinctrl-0 = <&i2s_bus>;
+	#sound-dai-cells = <0>;
+	status = "okay";
+};
+
+&i2s_8ch {
+	#sound-dai-cells = <0>;
+	status = "disabled";
 };
 
 &pwm0 {
-- 
2.35.3

